src/library/dao/playlistdao.cpp

```diff
@@ -900,7 +898,8 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
 
         for (int limit = 10; limit > 0 && conflictFound; limit--) {
             int randomShuffleSetIndex =
-                    (int)(qrand() / (RAND_MAX + 1.0) * (newPositions.count()));
+                    static_cast<int>(QRandomGenerator::global()->generate() /
+                            (UINT_MAX + 1.0) * (newPositions.count()));
```

Comment:
I didn't quite what this is supposed to do... please double check if this is correct. Could this overflow?

Reply:
you van use 
static_cast<int>(QRandomGenerator::global()->generateDouble() * newPositions.count());